### PR TITLE
QueryJetpackConnection: Make the siteId prop be optional

### PR DESCRIPTION
**Pull request by oskosk**
* Updates the `QueryJetpackConnection` component to not declare its `siteId` prop as required.
* Updates the component to only request data if the `siteId` prop is not `null`.

### CODE REVIEW

File: client/components/data/query-jetpack-connection/README.md [hunk 1]
```diff
@@ -28,7 +28,7 @@ export default function MyJetpackConnectionStatus( { jetpackConnection } ) {
 
 <table>
 	<tr><th>Type</th><td>Number</td></tr>
-	<tr><th>Required</th><td>Yes</td></tr>
+	<tr><th>Required</th><td>No</td></tr>
```
Comment (tyxla) [inline client/components/data/query-jetpack-connection/README.md]: Let's keep this one "Yes". Although the `siteId` can be `null` and it's not required in the context of `PropTypes`, this query component can't function without it.

File: client/components/data/query-jetpack-connection/index.jsx [hunk 2]
```diff
@@ -12,13 +12,15 @@ import { requestJetpackConnectionStatus } from 'state/jetpack/connection/actions
...
 	static propTypes = {
-		siteId: PropTypes.number.isRequired,
+		siteId: PropTypes.number,
 		requestingJetpackConnectionStatus: PropTypes.bool,
 		requestJetpackConnectionStatus: PropTypes.func
 	};
 
 	componentWillMount() {
-		this.request( this.props );
+		if ( this.props.siteId !== null ) {
```
Comment (tyxla) [inline client/components/data/query-jetpack-connection/index.jsx]: Let's do that check in `this.request` instead - see https://github.com/Automattic/wp-calypso/blob/master/client/components/data/query-jetpack-user-connection/index.jsx#L33 for example.

**Comment by Copons** [comment]
LGTM! 👍

**Comment by lancewillett** [comment]
Hi @Copons @oskosk — ready to merge? 29 days ago was the last update.

**Comment by alisterscott** [comment]
Are we still planning to merge this?

**Comment by tyxla** [comment]
This one covers just a small percentage of all occurrences. After a brief discussion with @oskosk on Slack, we decided to close this one for now, and we'll revisit in the future to cover more cases.

**Pull request closed without merging**